src/Engine.php

```diff
@@ -12,47 +12,23 @@
  */
 class Engine
 {
-
-    /**
-     *
-     * @var array
-     */
-    protected $tests = [];
-
-    /**
-     *
-     * @var Storage\StorageInterface
-     */
-    protected $storage;
-
-    /**
-     *
-     * @var Analytics\AnalyticsInterface
-     */
-    protected $analytics;
-
-    /**
-     *
-     * @param \Phpab\Phpab\Storage\StorageInterface $storage
-     * @param \Phpab\Phpab\Analytics\AnalyticsInterface $analytics
-     */
-    public function __construct(Storage\StorageInterface $storage, Analytics\AnalyticsInterface $analytics) {
-
-        $this->storage = $storage;
-
-        $this->analytics = $analytics;
-
+    public function __construct(Storage\StorageInterface $storage, Analytics\AnalyticsInterface $analytics)
+    {
+        // TODO: implement method
     }
 
     /**
      *
      * @param \Phpab\Phpab\Test $test
      */
-    public function addTest(Test $test) {
-
-        //@todo: TBD: This should avoid running the same test twice and have unexpected results
-        $this->tests[$test->getName()] = $test;
+    public function addTest(Test $test)
```

Comment:
At the time of writing that example we didn't see any use case for test interfaces, since they were pure containers.